Dockerfile

```diff
@@ -0,0 +1,845 @@
+FROM debian:buster-20180213
+RUN apt-get -y update \
+    && apt-get -y install --no-install-recommends \
+    build-essential \
+    curl \
+    ca-certificates \
+    gawk \
+    m4 \
+    xz-utils
+ENV ARCH x86_64
+ENV SYS linux
+ENV VENDOR talos
+ENV SYSROOT /${VENDOR}
+ENV TOOLCHAIN /toolchain
+ENV TARGET ${ARCH}-${VENDOR}-${SYS}-gnu
+ENV SRC /usr/local/src
+ENV ROOTFS /rootfs
+ENV PATH ${TOOLCHAIN}/bin:$PATH
+RUN mkdir -p ${SYSROOT}${TOOLCHAIN}
+RUN ln -s ${SYSROOT}${TOOLCHAIN} ${TOOLCHAIN}
+RUN mkdir -p ${TOOLCHAIN}/lib
+RUN ln -s lib ${TOOLCHAIN}/lib64
+WORKDIR ${SRC}
+
+ENV CURL "curl --retry 5 --retry-delay 0 --retry-max-time 45 -L"
+
+# binutils pass 1
+
+WORKDIR ${SRC}/binutils-pass-1/build
+RUN ${CURL} https://ftp.gnu.org/gnu/binutils/binutils-2.29.1.tar.xz | tar --strip-components=1 -xJ
+WORKDIR ${SRC}/binutils-pass-1/build/build
+RUN ../configure \
+    --prefix=${TOOLCHAIN} \
+    --with-sysroot=${SYSROOT} \
+    --with-lib-path=${TOOLCHAIN}/lib \
+    --target=${TARGET} \
+    --disable-nls \
+    --disable-werror
+RUN make -j $(($(nproc) / 2))
+RUN make install
+
+# gcc pass 1
+
+WORKDIR ${SRC}/gcc-pass-1
+RUN ${CURL} https://ftp.gnu.org/gnu/gcc/gcc-7.3.0/gcc-7.3.0.tar.xz | tar --strip-components=1 -xJ \
```

Comment:
nit, probably want to parameterize these versions. env vars perhaps?

Reply:
👍 Yeah env vars probably make the most sense.

Reply:
@bradbeam How do you feel about me pushing this PR through as is, and I submit a followup PR. Reason being is that the builds take a long time, and I'd like to merge this in knowing it is building now. Took me quite some time to get to this point :D